core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -955,6 +955,27 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
         .set(EXECUTOR_ALLOW_SPARK_CONTEXT, true)).stop()
     }
   }
+
+  test("SPARK-33084: Add jar support ivy url") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local-cluster[3, 1, 1024]"))
+
+    // default transitive=false, only download specified jar
+    sc.addJar("ivy://org.apache.hive.hcatalog:hive-hcatalog-core:2.3.7")
+    assert(
+      sc.listJars().exists(_.contains("org.apache.hive.hcatalog_hive-hcatalog-core-2.3.7.jar")))
+
+    // test download ivy URL jar return multiple jars
+    sc.addJar("ivy://org.scala-js:scalajs-test-interface_2.12:1.2.0?transitive=true")
+    assert(sc.listJars().exists(_.contains("scalajs-library_2.12")))
+    assert(sc.listJars().exists(_.contains("scalajs-test-interface_2.12")))
+
```

Comment:
@maropu This case can show https://github.com/apache/spark/pull/29966#discussion_r529228750